ios/brave-ios/Sources/Brave/Frontend/Browser/Search/RecentSearchQRCodeScannerController.swift

```diff
@@ -142,12 +153,69 @@ extension RecentSearchQRCodeScannerController {
       $0.textColor = .braveLabel
     }
 
+    var scannedText: String? {
+      didSet {
+        let processedScannedResult = processScannedText()
+        let truncationMode = processedScannedResult.truncationMode
+
+        if let scannedText = processedScannedResult.string {
+          let paragraphStyle = NSMutableParagraphStyle()
+          paragraphStyle.lineBreakMode = truncationMode
+          paragraphStyle.baseWritingDirection = .leftToRight
+
+          let title = NSAttributedString(
+            string: scannedText,
+            attributes: [
+              .font: UIFont.preferredFont(forTextStyle: .body),
+              .paragraphStyle: paragraphStyle,
+            ]
+          )
+
+          scannedDisplayButton.titleLabel?.lineBreakMode = truncationMode
+          scannedDisplayButton.setAttributedTitle(title, for: .normal)
+        } else {
+          scannedDisplayButton.setTitle(nil, for: .normal)
+        }
+      }
+    }
+
+    private(set) lazy var scannedDisplayButton = UIButton().then {
+      $0.contentEdgeInsets = UIEdgeInsets(top: 12, left: 16, bottom: 12, right: 16)
+      $0.titleLabel?.font = UIFont.preferredFont(forTextStyle: .body)
+      $0.titleLabel?.lineBreakMode = .byTruncatingTail
+      $0.setTitleColor(.white, for: .normal)
+      $0.backgroundColor = UIColor(braveSystemName: .primitivePurple60)
```

Comment:
Should be `buttonBackground`

Reply:
If this button isnt themed for some reason you can ignore this and the comment below